test/unit/firebase-app.spec.ts

```diff
@@ -149,6 +155,70 @@ describe('FirebaseApp', () => {
+      expect(app.options.projectId).to.equal("hipster-chat-mock");
+      expect(app.options.storageBucket).to.equal("hipster-chat.appspot.mock");
+    });
+
+    it('should throw when the environment variable points to non existing file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_non_existant.json';
+      
+      expect(() => {
+       const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      }).to.throw(`Failed to read app options file: Error: ENOENT: no such file or directory, open './test/resources/firebase_config_non_existant.json'`);
+    });
+
+    it('should throw when the environment variable points to non parsable file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_bad.json'  ;
+      expect(() => {
+        const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      }).to.throw(`Failed to parse app options file`);
+    });
+
+    it('should use the existing values when some of them exist', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptions, mocks.appName);
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.projectId).to.equal('hipster-chat-mock');
+      expect(app.options.storageBucket).to.equal('bucketName.appspot.com');
+    });
+
+    it('should use the existing values when some of them are not overwritten', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_partial.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsAuthDB, mocks.appName);
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.projectId).to.equal('hipster-chat-mock');
+      expect(app.options.storageBucket).to.equal(undefined);
+    });
+
+    it('should not throw when the config environment variable is not set', () => {
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      expect(app.options.databaseURL).to.equal(undefined);
+      expect(app.options.projectId).to.equal(undefined);
+      expect(app.options.storageBucket).to.equal(undefined);
+    });
+
+    it('should init with no params setting default credential', () => {
+      const app = firebaseNamespace.initializeApp();
+      expect(app.options.credential).to.be.instanceOf(ApplicationDefaultCredential)
+      expect(app.options.databaseURL).to.equal(undefined);
+      expect(app.options.projectId).to.equal(undefined);
+      expect(app.options.storageBucket).to.equal(undefined);
+    });
+
+    it('should init with params from the config file, setting default credential', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config.json';
```

Comment:
Need one more test case for when options are not specified, and neither the env variable is set.